Build failure with ShellCheck 0.4.7

- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

A successful hadolint 1.2.4 build with ShellCheck 0.4.7.

### Actual behavior

A failed build.

### Steps to reproduce the behavior

brew pull https://github.com/Homebrew/homebrew-core/pull/21831
brew install -dvs hadolint

Full build log: https://gist.github.com/ilovezfs/4b4ca3d209780585d704c1636bf0d778

The problem does not occur with ShellCheck 0.4.6.

The error is

```
Failed to install hadolint-1.2.3
Build log ( /private/tmp/hadolint-20171217-99291-1437fs3/hadolint-1.2.4/.cabal-sandbox/logs/ghc-8.2.2/hadolint-1.2.3-Jp4PH93kGpG51V3rMmG4HI.log ):
cabal: Entering directory '.'
Configuring hadolint-1.2.3...
clang: warning: -Wl,-headerpad_max_install_names: 'linker' input unused [-Wunused-command-line-argument]
clang: warning: argument unused during compilation: '-L/usr/local/lib' [-Wunused-command-line-argument]
clang: warning: argument unused during compilation: '-L/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries' [-Wunused-command-line-argument]
Preprocessing library for hadolint-1.2.3..
Building library for hadolint-1.2.3..
[1 of 3] Compiling Hadolint.Bash    ( src/Hadolint/Bash.hs, dist/dist-sandbox-f3f6770f/build/Hadolint/Bash.o )

src/Hadolint/Bash.hs:8:81: error:
    • Couldn't match expected type ‘CheckSpec’
                  with actual type ‘Maybe Shell -> CheckSpec’
    • Probable cause: ‘spec’ is applied to too few arguments
      In the second argument of ‘checkScript’, namely ‘spec’
      In the second argument of ‘($)’, namely ‘checkScript si spec’
      In the second argument of ‘($)’, namely
        ‘runIdentity $ checkScript si spec’
  |
8 | shellcheck bashScript = map comment $ crComments $ runIdentity $ checkScript si spec
  |                                                                                 ^^^^

src/Hadolint/Bash.hs:12:38: error:
    • Couldn't match expected type ‘Bool’ with actual type ‘[a0]’
    • In the third argument of ‘CheckSpec’, namely ‘exclusions’
      In the expression: CheckSpec filename script exclusions (Just Bash)
      In an equation for ‘spec’:
          spec = CheckSpec filename script exclusions (Just Bash)
   |
12 |     spec = CheckSpec filename script exclusions (Just Bash)
   |                                      ^^^^^^^^^^

src/Hadolint/Bash.hs:12:50: error:
    • Couldn't match expected type ‘[Integer]’
                  with actual type ‘Maybe Shell’
    • In the fourth argument of ‘CheckSpec’, namely ‘(Just Bash)’
      In the expression: CheckSpec filename script exclusions (Just Bash)
      In an equation for ‘spec’:
          spec = CheckSpec filename script exclusions (Just Bash)
   |
12 |     spec = CheckSpec filename script exclusions (Just Bash)
   |                                                  ^^^^^^^^^
cabal: Leaving directory '.'
cabal: Error: some packages failed to install:
hadolint-1.2.3-Jp4PH93kGpG51V3rMmG4HI failed during the building phase. The
exception was:
ExitFailure 1
```

--- a/hadolint.cabal
+++ b/hadolint.cabal
@@ -23,7 +23,7 @@ library
                      , bytestring >=0.10
                      , split >= 0.2
                      , language-docker >= 1.0.0
-                     , ShellCheck >= 0.4.6
+                     , ShellCheck >= 0.4.7
   default-language:    Haskell2010
 
 executable hadolint
@@ -49,7 +49,7 @@ test-suite hadolint-unit-tests
                      , bytestring >=0.10
                      , split >= 0.2
                      , language-docker >= 1.0.0
-                     , ShellCheck >= 0.4.6
+                     , ShellCheck >= 0.4.7
                      , HUnit >= 1.2
                      , hspec
                      , hadolint

--- a/src/Hadolint/Bash.hs
+++ b/src/Hadolint/Bash.hs
@@ -9,7 +9,8 @@ shellcheck bashScript = map comment $ crComments $ runIdentity $ checkScript si 
   where
     comment (PositionedComment _ _ c) = c
     si = mockedSystemInterface [("", "")]
-    spec = CheckSpec filename script exclusions (Just Bash)
+    spec = CheckSpec filename script sourced exclusions (Just Bash)
     script = "#!/bin/bash\n" ++ bashScript
     filename = "" -- filename can be ommited because we only want the parse results back
+    sourced = False
     exclusions = []

--- a/stack.yaml
+++ b/stack.yaml
@@ -4,4 +4,5 @@ packages:
 - .
 extra-deps:
 - language-docker-1.0.0
+- ShellCheck-0.4.7
 resolver: lts-9.18

